File: javascript/09-stringly-typed/test/issue_updater.test.js
import assert from 'node:assert';
import test from 'node:test';
import { IssueUpdater } from '../src/issue_updater.js';

test('closes issue', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(42, 'close');
  assert.ok(result.includes('status=closed'));
});

test('opens issue', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(42, 'open');
  assert.ok(result.includes('status=open'));
});

test('sets in_progress', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(42, 'progress');
  assert.ok(result.includes('status=in_progress'));
});

test('sets priority 1', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(42, 'close:1');
  assert.ok(result.includes('priority=1'));
});

test('sets priority 2', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(42, 'open:2');
  assert.ok(result.includes('priority=2'));
});

test('defaults to priority 3', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(42, 'progress');
  assert.ok(result.includes('priority=3'));
});

test('includes issue id', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(99, 'close');
  assert.ok(result.includes('Issue 99'));
});

test('ignores invalid priority', () => {
  const updater = new IssueUpdater();
  const result = updater.update_issue(42, 'close:99');
  assert.ok(result.includes('priority=3'));
});

File: ruby/09-stringly-typed/test/issue_updater_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative '../src/issue_updater'

class IssueUpdaterTest < Minitest::Test
  def setup
    @updater = IssueUpdater.new
  end

  def test_closes_issue
    result = @updater.update_issue(42, 'close')
    assert_includes result, 'status=closed'
  end

  def test_opens_issue
    result = @updater.update_issue(42, 'open')
    assert_includes result, 'status=open'
  end

  def test_sets_in_progress
    result = @updater.update_issue(42, 'progress')
    assert_includes result, 'status=in_progress'
  end

  def test_sets_priority_1
    result = @updater.update_issue(42, 'close:1')
    assert_includes result, 'priority=1'
  end

  def test_sets_priority_2
    result = @updater.update_issue(42, 'open:2')
    assert_includes result, 'priority=2'
  end

  def test_defaults_to_priority_3
    result = @updater.update_issue(42, 'progress')
    assert_includes result, 'priority=3'
  end

  def test_includes_issue_id
    result = @updater.update_issue(99, 'close')
    assert_includes result, 'Issue 99'
  end

  def test_ignores_invalid_priority
    result = @updater.update_issue(42, 'close:99')
    assert_includes result, 'priority=3'
  end
end
